core/channel/migrate_channel.hpp

```diff
@@ -29,40 +29,27 @@ namespace husky {
 using base::BinStream;
 
 template <typename ObjT>
-class MigrateChannel : public ObjList2ObjListChannel<ObjT, ObjT> {
+class MigrateChannel : public ChannelBase {
    public:
-    MigrateChannel(ObjList<ObjT>* src, ObjList<ObjT>* dst) : ObjList2ObjListChannel<ObjT, ObjT>(src, dst) {
-        this->src_ptr_->register_outchannel(this->channel_id_, this);
-        this->dst_ptr_->register_inchannel(this->channel_id_, this);
-    }
-
-    ~MigrateChannel() override {
-        this->src_ptr_->deregister_outchannel(this->channel_id_);
-        this->dst_ptr_->deregister_inchannel(this->channel_id_);
-    }
+    MigrateChannel() = default;
 
     MigrateChannel(const MigrateChannel&) = delete;
     MigrateChannel& operator=(const MigrateChannel&) = delete;
 
     MigrateChannel(MigrateChannel&&) = default;
     MigrateChannel& operator=(MigrateChannel&&) = default;
 
-    void customized_setup() override { migrate_buffer_.resize(this->worker_info_->get_largest_tid() + 1); }
+    void buffer_setup() { migrate_buffer_.resize(this->worker_info_->get_largest_tid() + 1); }
 
     void migrate(ObjT& obj, int dst_thread_id) {
-        auto idx = this->src_ptr_->delete_object(&obj);
+        auto idx = this->obj_list_ptr_->delete_object(&obj);
         migrate_buffer_[dst_thread_id] << obj;
-        this->src_ptr_->migrate_attribute(migrate_buffer_[dst_thread_id], idx);
+        this->obj_list_ptr_->migrate_attribute(migrate_buffer_[dst_thread_id], idx);
     }
 
-    void prepare() override {}
+    void set_obj_list(ObjList<ObjT>* obj_list_ptr) { obj_list_ptr_ = obj_list_ptr; }
```

Comment:
The channel does not depend on `dst_list` to `send`. So I think there is no need for `set_dst_obj_list`. Anyway, changing the name to `set_src_obj_list` makes the code clearer.